### ISSUE TITLE
Ignite X web crashes

### ISSUE THREAD

**Issue by lindboe** [infinitered/ignite#2807]
### Describe the bug

Generate a new project and run it on web.

```
ReferenceError: setImmediate is not defined
KeyboardAwareScrollView<@http://localhost:8081/node_modules/expo/AppEntry.bundle?platform=web&dev=true&hot=false&lazy=true&transform.engine=hermes&transform.routerRoot=app:165874:9
ScreenWithScrolling@http://localhost:8081/node_modules/expo/AppEntry.bundle?platform=web&dev=true&hot=false&lazy=true&transform.engine=hermes&transform.routerRoot=app:139643:9
div
View<@http://localhost:8081/node_modules/expo/AppEntry.bundle?platform=web&dev=true&hot=false&lazy=true&transform.engine=hermes&transform.routerRoot=app:34588:21
KeyboardAvoidingView@http://localhost:8081/node_modules/expo/AppEntry.bundle?platform=web&dev=true&hot=false&lazy=true&transform.engine=hermes&transform.routerRoot=app:54929:7
div
```

<img width="1144" alt="Screenshot 2024-10-16 at 10 36 00 AM" src="https://github.com/user-attachments/assets/67e04475-d4db-47a7-a9ee-338a52db2caf">

It looks like the addition of `KeyboardAwareScrollView` isn't compatible with web, and we need to fall back to regular `ScrollView` where it's used.

We should probably also have a smoke test for web to make sure it's able to load without errors, but if it's our first web test that might be more scope when we just want this fixed, we can break that out into another task.

### Ignite version

10.0.0

### Additional info

```
System
  platform           darwin
  arch               arm64
  cpu                10 cores     Apple M1 Max
  directory          TextCheck    /Users/lizzi/dev/igniteTests/TextCheck

JavaScript (and globally-installed packages)
  node                22.6.0                                    /opt/homebrew/bin/node
  npm                 10.8.2                                    /opt/homebrew/bin/npm
    expo-cli          6.0.8
    npm               10.8.2
    yarn              1.22.17
  yarn                1.22.17                                   /opt/homebrew/bin/yarn
    create-expo-app   3.0.0
    create-vite       4.1.0
  pnpm                9.4.0                                     /opt/homebrew/bin/pnpm
                      <no pnpm global package info available>
  bun                 1.1.21                                    /opt/homebrew/bin/bun
  expo                51.0.37                                   managed
  expo global cli     6.0.8                                     Deprecated: Found 'expo-cli' installed. Please remove it.

Ignite
  ignite-cli         10.0.0       /Users/lizzi/.npm/_npx/e31027f3785124a8/node_modules/.bin/ignite
  ignite src         build        /Users/lizzi/.npm/_npx/e31027f3785124a8/node_modules/ignite-cli/build

Android
  java               17.0.11      /Users/lizzi/.jenv/shims/java
  android home       -            /Users/lizzi/Library/Android/sdk

iOS
  xcode              16.0
  cocoapods          1.13.0       /Users/lizzi/.rbenv/shims/pod

Tools
  git                git version 2.46.0   /opt/homebrew/bin/git
```

**Comment by frankcalise** [comment]
That's interesting because it looks like web was stubbed out a bit ago: https://github.com/kirillzyusko/react-native-keyboard-controller/issues/208

**Comment by frankcalise** [comment]
Also interesting: https://github.com/necolas/react-native-web/issues/1891#issuecomment-887502705

**Comment by frankcalise** [comment]
An open issue at react-navigation about the useLayoutEffect + setOptions: https://github.com/react-navigation/react-navigation/issues/11748

**Comment by fnandoz21** [comment]
Still getting the same issue that was raised above. Any timeline on the fix?

**Comment by jaslam-ca** [comment]
Can this be fixed quickly. I depend on web development and kind of stuck on it.

**Comment by improwise** [comment]
I am having the same problem. It does seem to work when selecting to use Expo Router in setup though (even though if we do, almost all of the demo material is gone even if we select to keep it).

**Comment by cgouguen** [comment]
Same problem here. To me Ignite X is unusable unfortunately, I have to stick with Ignite 9 until this is fixed...

**Comment by maryam-khan-dev** [comment]
Same issue here :(

**Comment by frankcalise** [comment]
Closed by #2856 and fixed in 10.1.0 which is now released
